src/sentry/event_manager.py

```diff
@@ -477,29 +477,6 @@ def save(self, project_id, raw=False, assume_normalized=False):
             id=project.organization_id
         )
 
-        # Check to make sure we're not about to do a bunch of work that's
```

Comment:
Please let's add some doc on the expected behavior so that none thinks this is a bug one year from now.

Also, this can have side effects we did not anticipate. Please gate this logic with a feature flag or anything that allows you to roll it out partially or turn it off quickly in case some issue comes up we did not think about.

Reply:
I added this behind an option (it's just on/off for all events), not sure if it's worth gating this for a subset of accounts first